### PR TITLE
get secrets from auth file from s3

### PR DESCRIPTION
**Pull request by vasukulkarni**
store our auth details in simple s3 bucket and read secrets from dict.

### CODE REVIEW

**Comment by vasukulkarni** [comment]
@prsurve FYI, we can push IPMI details here

**Comment by vasukulkarni** [comment]
ex: ``` Config(AUTH={'quay': {'access_token': 'ocs-ci-auth'}, 'ipmi': [{'hostname': 'hostname', 'console': 'console', 'username': 'username', 'password': 'password'}]}```

File: ocs_ci/utility/aws.py [hunk 1]
```diff
@@ -1094,3 +1096,23 @@ def check_root_volume(volume):
...
     """
     return True if volume['attachments'][0]['DeleteOnTermination'] else False
+
+
+def get_auth_dict(bucket_name=constants.BUCKETNAME, filename=constants.AUTHYAML):
+    """
+    Get the auth file that has secrets from the S3 and return as dict
+
+    Args:
+        bucket_name(string): name of the bucket
```
Comment (vavuthu) [inline ocs_ci/utility/aws.py:1106]: space between bucket_name and (

File: ocs_ci/utility/aws.py [hunk 2]
```diff
@@ -1094,3 +1096,23 @@ def check_root_volume(volume):
...
+        filename(string): name of the file in bucket
```
Comment (vavuthu) [inline ocs_ci/utility/aws.py:1107]: s/filename(string)/filename (string)/

File: ocs_ci/utility/aws.py [hunk 3]
```diff
@@ -1094,3 +1096,23 @@ def check_root_volume(volume):
...
+
+    Returns:
+        dict: returns the auth file as python dict
```
Comment (vavuthu) [inline ocs_ci/utility/aws.py:1110]: leave blank line below

File: ocs_ci/ocs/constants.py [hunk 4]
```diff
@@ -200,6 +200,10 @@
 OSD_PVC_GENERIC_LABEL = "ceph.rook.io/DeviceSet"
 CEPH_ROOK_IO_PVC_LABEL = 'ceph.rook.io/pvc'
 
+# Auth Yaml
+BUCKETNAME = 'ocs-ci-data'
```
Comment (MeridianExplorer) [inline ocs_ci/ocs/constants.py:204]: I think it might be better to give this constant a more specific name, like `OCSCI_DATA_BUCKET`

Comment (vasukulkarni) [reply ocs_ci/ocs/constants.py]: all review comment fixed.

**Pull request merged**